### PR TITLE
【CPU】change Batch norm reserve_space dtype

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR Category

Execute Infrastructure

### PR Types

Bug fixes

### Description

pcard-67164

由于cpu/xpu kernel 没有对bacthnorm 的reserve_space 做正确的类型处理，导致传递的结果有误，在下一个cast kernel 中报类型不符的报错

### CODE REVIEW

File: paddle/phi/kernels/xpu/batch_norm_kernel.cc [hunk 1]
```diff
@@ -98,7 +98,7 @@ void BatchNormKernel(const Context& dev_ctx,
   dev_ctx.template Alloc<float>(variance_out);
   dev_ctx.template Alloc<float>(saved_mean);
   dev_ctx.template Alloc<float>(saved_variance);
-
+  dev_ctx.template Alloc<uint8_t>(reserve_space);
```
Comment (xiaoguoguo626807) [inline paddle/phi/kernels/xpu/batch_norm_kernel.cc]:
```suggestion
```

**Review by changeyoung98 — APPROVED** [review]
LGTM
